Microsoft/edx-configuration
# sync edx-configuration with ficus.1

#### What does this PR do? Please provide some context

we forked /edx/configuration about a dozen commits before ficus.1 . This change cherry-picks the missing commits into our repo. Functionally speaking:
- this is the first of three changes we need in order to harden MYSQL_ADMIN_PASSWORD for onebox deployments (aks: devstack and fullstack instances). FYI:
  - part two is https://github.com/Microsoft/edx-configuration/pull/43 and
  - part three is https://github.com/Microsoft/oxa-tools/compare/6e0f1aa05b529cfafc1e776dd05af4e986c289ae...451a2a479cb9ea85bfba3cba7836ab28eb99684b which is part of https://github.com/Microsoft/oxa-tools/pull/165
- this helps us better determine if bugs are introduced with our changes or part of the ficus.1 release by putting the fork at a well-established boundary 

#### Where should the reviewer start?

anywhere. this whole change is purely cherry-picks so feel free to take a look at the upstream commits (and corresponding pull requests) if you have concerns

#### How can this be manually tested? (brief repro steps)

After all three parts are merged then deploying onebox (without -b edx) will have strong mysql credentials . See https://github.com/Microsoft/oxa-tools/pull/165 for more details.

#### What are the relevant TFS items? (list id numbers)

100757

#### Definition of done:

#### Reminders BEFORE merging
1. Get at least two approvals
1. If you're merging from a feature branch into the development branch then "flatten" or "squash" commits
1. If merging from the development branch into master (or porting changes from upstream) then use github's UI to get review feedback, but use the git command line interface to complete the actual merge.

#### Reminders AFTER merging
1. Delete the remote feature branch
1. Resolve relevant TFS items
1. (reverse merge) If you merged from the development branch into master then check to see if there are any changes in master that can be merged down to the development branch (like hotfixes, etc). In this case, use github's UI for feedback and the git command line interface for the actual merge.

[//]: # ( todo: If you merged into development branch then verify change in our "rolling deployment" environment. Then notify stakeholders interested in or involved with the change )

[//]: # ( fyi: This content was heavily inspired by )
[//]: # ( 1 Our team's policies and processes )
[//]: # ( 2 https://github.com/sprintly/sprint.ly-culture/blob/master/pr-template.md )
[//]: # ( 3 The book "The Checklist Manifesto: How to Get Things Right" by Atul Gawande )
[//]: # ( 4 https://github.com/Azure/azure-event-hubs/blob/master/.github/PULL_REQUEST_TEMPLATE.md )


Configuration Pull Request
---
#### (For changes proposed to upstream)
Make sure that the following steps are done before merging
